chapters/sampling-randomization-power.tex

```diff
@@ -1,266 +1,615 @@
+
+%%%%% IV regression
+
+\textbf{Instrumental variables (IV)}\sidenote{
+	\url{https://dimewiki.worldbank.org/Instrumental\_Variables}}
+\index{instrumental variables}
+designs, unlike the previous approaches,
+assume that the treatment effect is not directly identifiable.
+Similar to RD designs,
+IV designs focus on a subset of the variation in treatment take-up.
+Where RD designs use a ``sharp'' or binary cutoff,
+IV designs are ``fuzzy'', meaning that the input does not completely determine
+the treatment status, but instead influence the \textit{probability of treatment}.
+
+You will need variables in your data
+that can be used to estimate the probability of treatment for each unit.
+These variables are called \textit{instruments}.
+Testing that a variable is a valid instrument is a non-trivial and important task
+that is outside the scope of this book.
+In IV designs, instead of the ordinary regression estimator,
+a special version called two-stage least squares (2SLS) is used
+to estimate the impact of the treatment.
+Stata has a built-in command called \texttt{ivregress},
+and another popular implementation is the user-written command \texttt{ivreg2}.
+
+%%%%% Matching
+
+\textbf{Matching}\sidenote{
+	\url{https://dimewiki.worldbank.org/Matching}}
+methods use observable characteristics to construct
+sets or pairs of treatment and control units
+so that the observations in each set are as similar as possible. \index{matching}
+These sets can either consist of exactly one treatment and one control observation (one-to-one),
+or it can be a set observations where either
+both groups have more than one observation represented (many-to-many),
+or where only one group has more than one observation included (one-to-many).
+
+The matching can even be done before the random assignment,
+so that treatment can be randomized within each matching set.
+This is a type of experimental design.
+Furthermore, if no control observations were identified before the treatment,
+then matching can be used to ex-post identify a control group,
+by finding the observations that are the most similar to the treated group.
+It is very difficult to test the validity of an ex-post matching
+as one would have to show that the difference in outcome is
+due to the impact of the treatment and not due to the groups not being similar enough.
+
+A valid matching must be made on data that is not related to the treatment
+or anything that the treatment could have affected in any way.
+Many matching algorithms can only match on a single variable,
+so you first have to turn many variables into a single varaible
+by using an index or a propensity score.\sidenote{
+	\url{https://dimewiki.worldbank.org/Propensity\_Score\_Matching}}
+The \texttt{iematch}\sidenote{
+	\url{https://dimewiki.worldbank.org/iematch}}
+command in the \texttt{ietoolkit} package developed by DIME Analytics
+produces matchings based on a single continuous matching variable.
```

Comment:
Isn't that explained in the text? On most places where we mention `ietoolkit` commands we first say "you should do this" and then "this command does that", and I think that is what we are doing here as well. Is this case different?

Reply:
there isn't anything in the text about the gap that we tried to fill with ietoolkit. in general i think that's an important part of the pitch for any of these tools. that may be as simple as saying something like there was not a tool to produce matches based on single continuous variable, which is a common use case in DIME. Analytics built iematch to meet that need.

Reply:
I like your suggestion had there been neat origin stories for the `ietoolkit` commands. But it is usually not the case. For example, for `iematch` I had some code that solved this neatly that I kept reusing and modifying each time until it was almost a command, and then I did the last programming needed to make it a command. There was never the case that I did a thorough  search for a command that solved this. I did look until I felt I started to spend more time looking that just writing it myself. I think that is the most common case for most commands. 

It has rarely been the case we started with an empty do-file to fill a need that we saw.  We can say that we did that, but it would not be the truth. And I do not know if there was a command  out there that I did not find immediately the first time I programmed my own matching code that became the embryo for `iematch`.

Reply:
Ok. My assumption was that it wouldn't have made sense to spend time developing our own commands without doing some survey of what was out there and whether some other tool already met the need.

Reply:
We kind of did that, but not in one big sitting, it is a much more dynamic process over time than that. Each time before adding to my own code it was preceded by searching for an already existing command. If I had found such a command in a quick search, I would not have created my own code. Each time I or someone else wanted to reuse the code snippet it always started with a quick search, but since any conclusive search that the additional feature did not exist in some other command would take longer time than just adding that feature, I never did that conclusive survey of what was out there. I am pretty convinced that the sum of all quick searches means it does not exists, but I do not know for sure as it is a much more dynamic process than that. Almost none of these commands started with a blank ado-file with the intention of writing a command, and that is why there was never a survey.

Reply:
suggested rewording: 
DIME Analytics developed a command to match observations based on this single continuous variable: \texttt{iematch}\sidenote{
	\url{https://dimewiki.worldbank.org/iematch}}, part of the \texttt{ietoolkit} package.

Reply:
Implemented in 43d9dbf1cce12cf8222915b0aee0480fa8153673